README.md

```diff
@@ -210,3 +211,261 @@ developer console](https://code.google.com/apis/console/).
 ### Format of the payload
 
 This is the same format as [the reminder from the reminders API](#get-remindersid).
+
+## API users
+
+This API manages users.
+
+### POST `/users`
+
+This creates a new user.
+
+#### Input
+
+All properties must be present. There is no default value. For example:
+```json
+{
+  "forename": "Sherlock",
+  "email": "Sherlock.Holmes@baker-street.co.uk",
+  "password": "MoriartyShallReturn"
+}
+```
+
+Note that the password is stored in database using a modern hash algorithm
+(Argon2 currently).
+
+Currently the user is created as-is. In the future we'll implement a 2-step
+creation mechanism where the email would be used.
+
+#### Output
+* 201 if request succeeded, with `location` header indicating the new resource URL.
+* 400 if some properties are missing or invalid.
+
+### GET `/users/{id}`
+
+Required: the header `Authorization` that identifies a user. Only users
+belonging in the same group(s) can access information about another user.
+
+This returns information about a user.
+
+#### Output
+* 200 if request succeeded, with body:
+```json
+{
+  "forename": "Sherlock",
+  "email": "Sherlock.Holmes@baker-street.co.uk"
+}
+```
+* 404 if no user with this ID exists or if the logged-in user is not in a common
+  group.
+
+### GET `/users/{id}/groups`
+Required: the header `Authorization` that identifies a user. Only a user can see
+his own groups.
+
+This returns information about groups a user belongs to.
+
+#### Output
+* 200 if request succeeded, with body:
+```json
+[
+  {
+    "id": 1,
+    "name": "Holmes",
+    "location": "https://server.name/api/v1/groups/1",
+    "isAdmin": "true"
+  },
+  {
+    ...
+  }
+]
+```
+  `isAdmin` is `true` if this user is an admin of this group.
+* 404 if no user with this ID exists or if this user is not the logged-in user.
+
+### GET `/users/{id}/relations`
+Required: the header `Authorization` that identifies a user. Only a user can see
+his own relations.
+
+This returns all users this user has relations with through at least 1 group.
+
+#### Output
+* 200 if request succeeded, with body:
+```json
+[
+  {
+    "id": 2,
+    "forename": "James",
+    "email": "James.Moriarty@ReichenbachFalls.co.uk",
+    "commonGroups": ["Smart_People"],
+    "location": "https://server.name/api/v1/users/2"
+  },
+  {
+    ...
+  }
+]
+```
+* 404 if no user with this ID exists or if this user is not the logged-in user.
+
+
+### PUT `/users/{id}`
+
+Required: the header `Authorization` that identifies a user. A user can only be
+modified by himself.
+
+#### Input
+
+All properties must be present. For example:
```

Comment:
What if the user wants to change his name but not his password? Can we make the password fields optional?